### DISCUSSION TITLE
How to perfom DFT calculation with IntegralSolverPySCF?

### DISCUSSION THREAD

**Discussion by NanaEngo** [goodchemistryco/Tangelo#407] [Q&A] [answered]
Is there any way to perform DFT calculation with IntegralSolverPySCF?

**Comment by alexfleury-sb** [comment]
Hello @NanaEngo, this feature is not yet implemented. `IntegralSolverPySCF` is meant to compute one- and two-body integrals from a mean-field approximation, to eventually generate a second-quantized Hamiltonian. Since DFT deals with electronic density, it would require another design for `IntegralSolver`, which would work in the first quantization framework.

There are some work aiming at mapping DFT equations on quantum device, such as [Senjean et al. (2022)](https://arxiv.org/abs/2204.01443) and [Ko et al. (2023)](https://arxiv.org/abs/2307.07067). However, if your goal is to solely compare energies / densities with a classical DFT solution, it is possible to [use directly PySCF](https://pyscf.org/user/dft.html) (here I am assuming PySCF is installed as your question refers to `IntegralSolverPySCF`).

**Reply by NanaEngo** [reply]
Thank you for your prompt reply.
Precisely, it is the VQE calculation from the DFT that interests me. And my hope was to take advantage of the different modules already existing in Tangelo

**Reply by alexfleury-sb** [reply]
Hello again @NanaEngo, are you basing your idea on a published paper? If not, what would be the envisioned workflow?

**Comment by NanaEngo** [comment]
Hello Alexandre,

Your idea is base on
1. https://arxiv.org/abs/2302.03052
2. http://arxiv.org/abs/2312.12386v3
And we want to calculate Linear response properties of some OLED molecules
(large organic molecules).

*Nana Engo, PhD*
Director of Students Affairs
University of Buea

**Reply by alexfleury-sb** [reply, marked as answer]
After a quick glance at the papers, I can say the [SA_OO_Solver](https://github.com/sandbox-quantum/Tangelo/blob/main/tangelo/algorithms/variational/sa_oo_vqe_solver.py) would be useful to compute excited with orbital optimization in your case (as shown in the second paper). 

For the Hamiltonian definition in a VQE-in-DFT framework as described in the first reference, there is no complete workflow to do that in Tangelo. I suggest using directly PySCF as described in an earlier response to get integrals or a fermionic Hamiltonian, and then convert it into a tensor product of Pauli words with our toolset.
